src/utils/vcs.rs

```diff
@@ -249,6 +249,40 @@ pub fn git_repo_head_ref(repo: &git2::Repository) -> Result<String> {
     }
 }
 
+pub fn git_repo_base_ref(repo: &git2::Repository, remote_name: &str) -> Result<Option<String>> {
+    // Get the current HEAD commit
+    let head_commit = repo.head()?.peel_to_commit()?;
+
+    // Try to find the remote tracking branch
+    let remote_branch_name = format!("refs/remotes/{remote_name}/HEAD");
+    let remote_ref = repo.find_reference(&remote_branch_name).map_err(|e| {
+        anyhow::anyhow!(
+            "Could not find remote tracking branch for {}: {}",
+            remote_name,
+            e
+        )
+    })?;
+
+    find_merge_base_ref(repo, &head_commit, &remote_ref)
+}
+
+fn find_merge_base_ref(
+    repo: &git2::Repository,
+    head_commit: &git2::Commit,
+    remote_ref: &git2::Reference,
+) -> Result<Option<String>> {
```

Comment:
From what I can tell, we are never returning `Ok(None)`, so I think we can make this function return `Result<String>` instead

Reply:
Assuming I am correct, we should also be able to make `git_repo_base_ref` return a `Result<String>`